webapp/channels/src/packages/mattermost-redux/src/actions/status_profile_polling.ts

```diff
@@ -7,160 +7,85 @@ import type {UserProfile} from '@mattermost/types/users';
-
-const MAX_USER_IDS_PER_STATUS_REQUEST = 200; // users ids per 'users/status/ids'request
-const MAX_USER_IDS_PER_PROFILES_REQUEST = 100; // users ids per 'users/ids' request
-
-const pendingUserIdsForStatuses = new Set<string>();
-const pendingUserIdsForProfiles = new Set<string>();
-
-let intervalIdForFetchingPoll: NodeJS.Timeout | null = null;
-
-type UserIdsSingleOrArray = Array<UserProfile['id']> | UserProfile['id'];
-type AddUserIdsForStatusAndProfileFetchingPoll = {
-    userIdsForStatus?: UserIdsSingleOrArray;
-    userIdsForProfile?: UserIdsSingleOrArray;
-}
+import type {ActionFunc, ActionFuncAsync, ThunkActionFunc} from 'mattermost-redux/types/actions';
+import {IntervalDataLoader} from 'mattermost-redux/utils/data_loader';
 
 /**
- * Adds list(s) of user id(s) to the status and profile fetching poll. Which gets fetched based on user interval polling duration
- * Do not use if status or profile is required immediately.
+ * Adds list(s) of user id(s) to the status fetching poll. Which gets fetched based on user interval polling duration
+ * Do not use if status is required immediately.
  */
-export function addUserIdsForStatusAndProfileFetchingPoll({userIdsForStatus, userIdsForProfile}: AddUserIdsForStatusAndProfileFetchingPoll): ActionFunc<boolean> {
-    return (dispatch, getState) => {
-        function getPendingStatusesById() {
-            // Since we can only fetch a defined number of user statuses at a time, we need to batch the requests
-            if (pendingUserIdsForStatuses.size >= MAX_USER_IDS_PER_STATUS_REQUEST) {
-                // We use temp buffer here to store up until max buffer size
-                // and clear out processed user ids
-                const bufferedUserIds: string[] = [];
-                let bufferCounter = 0;
-                for (const pendingUserId of pendingUserIdsForStatuses) {
-                    if (pendingUserId.length === 0) {
-                        continue;
-                    }
-
-                    bufferedUserIds.push(pendingUserId);
-                    pendingUserIdsForStatuses.delete(pendingUserId);
-
-                    bufferCounter++;
-
-                    if (bufferCounter >= MAX_USER_IDS_PER_STATUS_REQUEST) {
-                        break;
-                    }
-                }
-
-                if (bufferedUserIds.length > 0) {
-                    dispatch(getStatusesByIds(bufferedUserIds));
-                }
-            } else {
-                // If we have less than max buffer size, we can directly fetch the statuses
-                const lessThanBufferUserIds = Array.from(pendingUserIdsForStatuses);
-                if (lessThanBufferUserIds.length > 0) {
-                    dispatch(getStatusesByIds(lessThanBufferUserIds));
-
-                    pendingUserIdsForStatuses.clear();
-                }
-            }
+export function addUserIdsForStatusFetchingPoll(userIdsForStatus: Array<UserProfile['id']>): ActionFunc<boolean> {
```

Comment:
It's unfortunate that plugins won't be able to take advantage of these semantics without updating their Redux library or copy/pasting all the supporting code for this thunk. Would we consider moving to a model where this action is a plain old javascript object, and the "things to load" are just stored in Redux itself? Then we could iterate and tinker on the implementation in the future (even switching to Sagas) without breaking the interface.

Reply:
That would be the idea with switching to Sagas since, once those are started, you just interact with those by dispatching plain actions. If we want to start using these in plugins, we could consider exposing these actions on the window or something similar, but I'd be hesitant about doing that until we've proven this system in the web app.

We're currently dealing with some issues because a web app change which we thought wouldn't affect anyone else broke the AI plugin, so I'm not wanting to encourage that for a bit 😅